web/src/app/admin/admin-message-logs/AdminMessageLogsGraph.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { useState } from 'react'
+import { useTheme } from '@mui/material/styles'
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  CardContent,
+  CardHeader,
+  Grid,
+  InputLabel,
+  MenuItem,
+  Paper,
+  Select,
+  Typography,
+} from '@mui/material'
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
+import {
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  LineChart,
+  Line,
+  Legend,
+} from 'recharts'
+import AutoSizer from 'react-virtualized-auto-sizer'
+import _ from 'lodash'
+import { DateTime, DateTimeFormatOptions } from 'luxon'
+import { useURLParam, useURLParams } from '../../actions'
+import Spinner from '../../loading/components/Spinner'
+import { gql, useQuery } from 'urql'
+
+type Stats = Array<{
+  start: string
+  end: string
+  count: number
+}>
+
+interface MessageLogGraphData {
+  date: string
+  label: string
+  count: number
+}
+
+const statsQuery = gql`
+  query messageStatsQuery(
+    $logsInput: MessageLogSearchOptions
+    $statsInput: TimeSeriesOptions!
+  ) {
+    messageLogs(input: $logsInput) {
+      stats {
+        timeSeries(input: $statsInput) {
+          start
+          end
+          count
+        }
+      }
+    }
+  }
+`
+
+export default function AdminMessageLogsGraph(): JSX.Element {
+  const theme = useTheme()
+
+  // graph duration set with ISO duration values, e.g. PT8H for a duration of 8 hours
+  const [duration, setDuration] = useURLParam<string>('graphInterval', 'PT1H')
+  const [now] = useState(DateTime.now())
+
+  const [{ search, start, end, graphInterval }] = useURLParams({
+    search: '',
+    start: now.minus({ hours: 8 }).toISO(),
+    end: now.toISO(),
+    graphInterval: 'PT1H',
+  })
+
+  const [{ data, fetching, error }] = useQuery({
+    query: statsQuery,
+    variables: {
+      logsInput: {
+        search,
+        createdAfter: start,
+        createdBefore: end,
+      },
+      statsInput: {
+        bucketDuration: graphInterval,
+        bucketOrigin: start,
+      },
+    },
+  })
+  const stats: Stats = data?.messageLogs?.stats?.timeSeries ?? []
+
+  const locale: DateTimeFormatOptions = {
+    month: 'short',
+    day: 'numeric',
+    year: 'numeric',
+    hour: 'numeric',
+    minute: 'numeric',
+  }
+  const graphData = React.useMemo(
+    (): MessageLogGraphData[] =>
+      stats.map(({ start, end, count }) => ({
+        count,
+        date: start,
+        label:
+          DateTime.fromISO(start).toLocaleString(locale) +
+          ' - ' +
+          DateTime.fromISO(end).toLocaleString(locale),
+      })),
+    [stats],
+  )
+
+  const formatIntervals = (label: string): string => {
+    // check for default bounds
+    if (label.toString() !== '0' && label !== 'auto') {
+      const dt = DateTime.fromFormat(label, 'MMM d, t')
```

Comment:
We can't parse locale strings -- they will change for different systems (e.g., locally, the page crashes for me since the format doesn't match) and are meant to be human-readable vs. standardized.

Can we update the label to be ISO-strings and then use this formatter to handle rendering them correctly?

Reply:
@mastercactapus The label is coming from the `recharts` `XAxis` component- I copied the code from the Alert Counts Graph.

I'll look into it though since it's crashing on your end.